assets/src/components/routeLadder.tsx

```diff
@@ -130,8 +91,11 @@ export const NewHeader = ({
               trigger="click"
               onShow={() => tagManagerEvent("alert_tooltip_clicked")}
             >
-              <div className="c-route-ladder__alert-icon">
-                <ExclamationTriangleFill aria-label="Route Alert" />
+              <div
+                className="c-route-ladder__alert-icon"
+                aria-label="Route Alert"
```

Comment:
Could you teach me why the `aria-label` needed moving?

Reply:
[This test](https://github.com/mbta/skate/blob/c64f1f1477f2b2b1466bdceec1b1c256c985a19e/assets/tests/components/routeLadder.test.tsx#L656) uses a [selector](https://github.com/mbta/skate/blob/c64f1f1477f2b2b1466bdceec1b1c256c985a19e/assets/tests/testHelpers/selectors/components/routeLadder.ts#L3) that tries to find the route alert icon using `byRole("generic", { name: "Route alert" })`. For reasons I don't fully understand, having the `aria-label` on the bootstrap icon doesn't actually put the aria-label onto the rendered component, but putting it on the wrapping `div` does.

I think this means that the new route ladder header route alert icon isn't doing accessibility quite right.

(FWIW - on `main` right now, the test is testing the old route ladder header, because the test group defaults to not being enabled, so the new route ladder header wasn't being tested for accessibility or an alert icon)

Reply:
But also, as I understand it, `aria-label` should be used as a last resort. This wasn't known when this was originally implemented, but a better way to do this would be to add the label text as a DOM node and linking it to the icon with `aria-labelledby` and have the text be visually hidden

Reply:
We could attempt to make things smarter, where if `aria-label` or `aria-labelledby` is provided, we could not set the [default `aria-hidden` value](https://github.com/mbta/skate/blob/c64f1f1477f2b2b1466bdceec1b1c256c985a19e/assets/src/helpers/bsIcons.tsx#L226), but I'm not sure it's worth it.

Reply:
> but a better way to do this would be to add the label text as a visually hidden DOM node and linking it to the icon with `aria-labelledby`

Want to either:
- pair on this?
- ...address as a follow-up?

Reply:
> Want to either:
> 
>     * pair on this?
> 
>     * ...address as a follow-up?

I'm good either way! Since it was old code, I'm also fine with how it is for now.

Reply:
I tried `aria-hidden={false}` and it still said `Unable to find an accessible element with the role "generic" and name "Route Alert"`

I also tried `aria-labelledby`, but that gave this error: 
```
Expected element to have accessible description:
  /Active Detour/i
Received:
```

PR (into this branch for now) below
- https://github.com/mbta/skate/pull/2703